esp/esp/users/controllers/usersearch.py

```diff
@@ -88,6 +89,12 @@ def query_from_criteria(self, user_type, criteria):
         else:
 
             ##  Select users based on all other criteria that was entered
+            if 'group' in criteria:
+                if criteria['group'] != "":
+                    group = criteria['group']
+                    Q_include &= Q(registrationprofile__user__groups__name=group)
```

Comment:
It should work to just do `Q(groups__name=group)`, `registrationprofile__user` should be unnecessary (and will make things slower in some cases).  (But see also below.)